### ISSUE TITLE
Segmentation support request for synapse and myelin sheath in EM data​

### ISSUE THREAD

**Issue by Yangxiao-fmmu** [computational-cell-analytics/micro-sam#1083]
**Context**  
I am using [micro_sam] to analyze electron microscopy (EM) data of neural tissues. The model successfully segments mitochondria with high accuracy, significantly accelerating my organelle analysis workflow. However, **two critical structures remain unidentifiable**:  
1. **Synapses** (presynaptic vesicles, postsynaptic densities)  
2. **Myelin sheaths** (multi-layered lipid-protein structures wrapping axons)  

**Current Capabilities**  
- ✅ **Mitochondria segmentation**: Works reliably on my EM datasets (example attached below).  
- ⚠️ **Morphometric parameters**: While segmentation is successful, I cannot export quantitative data such as:  
  - Object counts  
  - Perimeter length  
  - Cross-sectional area  
  - Myelin sheath thickness (if segmented)  

**Feature Requests**  
1. **Structural Segmentation Support**  
   - Is it technically feasible to extend segmentation to synapses and myelin sheaths?  
   - If yes, what training data or model adjustments would be required?  

2. **Quantitative Export Functionality**  
   - Add parameter output for segmented objects, including:  
     - Area (µm²)  
     - Perimeter (µm)  
     - Count per unit volume  
   - Preferred formats: CSV, Excel, or direct integration with Python (Pandas/NumPy)  

**Additional Notes**  
- Myelin sheaths exhibit unique ultrastructural features: concentric lipid layers separated by ~10–15 nm gaps).  
- Synapse identification may require detecting paired pre-/post-synaptic membranes and vesicle clusters.

**Comment by Yangxiao-fmmu** [comment]
Dear Dr. Constantinpape,

Thank you for your prompt and detailed response! I greatly appreciate your guidance on addressing my two challenges.

Regarding the ​​quantitative analysis​​:

I will explore the skimage.regionpropsapproach you recommended and test the Napari quantification plugin. As I'm still developing my Python skills, these resources will be valuable learning opportunities to extract morphometric parameters (area, perimeter, object counts) from my mitochondrial segmentations.

Regarding ​​synapse and myelin segmentation​​:

Thank you for suggesting Synapse-Net. I will evaluate this immediately. To better illustrate my data characteristics, I've attached two representative EM patches showing:

Presynaptic vesicles (circled) and postsynaptic densities (arrowheads) (DataSize=4608x3456, Magnification=20000)
Multi-layered myelin sheaths wrapping axons (DataSize=4608x3456, Magnification=7000)

<img width="1280" height="800" alt="Image" src="https://github.com/user-attachments/assets/2b911892-2234-4d0d-b4ea-458430a764e8" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/e10ee0e4-4755-42e0-9297-0509299d27a1" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/6325f6de-59a2-4a34-86f1-59fc5f016c58" />

<img width="2560" height="1600" alt="Image" src="https://github.com/user-attachments/assets/bc1a9830-f96a-4f8b-a028-40f1dbc21749" />

If convenient, I would sincerely appreciate your insights on:

Whether fine-tuning micro-SAM or using Synapse-Net would be more suitable for these structures
Any annotation best practices (e.g., key biomarkers to label)
Your expertise is invaluable to advancing this neuroanatomical research. Thank you for supporting the community with these powerful open-source tools!

                                                                                                                                                   Best regards,

                                                                                                                                                         Yangxiao

**Comment by constantinpape** [comment]
Hi @Yangxiao-fmmu ,
thank you for providing the details. Given what you shared I would try the following first:
- For synaptic vesicles: try the model `vesicles_2d` from SynapseNet. It could work for your data.
- For postsynaptic densities: SynapseNet does not support them (because we don't have data where they are so nicely stained). You can try to annotate them with microSAM. The best model for this is likely "Natural Images (SAM)". It is better for such structures than the EM model (which has specifically been trained for mitochondria and nuclei in EM). If this doesn't work you'd need to annotate them by hand (check out the napari label layer functionality for this). After you annotated densities in a few micrographs, you can then train a micro_sam model to automate the segmentation, see the notebook I linked for this earlier.
- Could you also share an image for myelin sheath and indicate how exactly you'd like to semgent them? Ideally by providing a zoom in and annotating it similar to how you did it for vesicles and PSD? The best annotation and segmentation strategy depends on which parts you want to segment.

**Comment by constantinpape** [comment]
Thank you for the explanation! You could give it a try with micro_sam, and again I would suggest to try with the `Natural Images (SAM)` model first. You could then try to finetune a model based on the annotations, as I outlined above. A general caveat here: SAM may struggle a bit with objects with "holes inside", but it's worth to give it a try. An alternative would be a more traditional U-Net based approach - but for this you would also need training data.
